src/utils/svgFactory.ts

```diff
@@ -1,45 +1,47 @@
-import fs, { promises as pfs } from 'fs'
+import fs from 'fs'
 import { customizedConfig, flipObject } from 'common/types'
 
 class SvgFactory {
     private svgCode: string = '';
     private pathToSvg: string;
-    private outputPathToSvg: string;
     private colorCode: string;
     private rotateAngle: number;
     private flip: flipObject;
+    private customized: boolean;
 
-    constructor (pathToSvg: string, outputPathToSvg: string, customizationConfig: customizedConfig) {
+    constructor (pathToSvg: string, customizationConfig: customizedConfig, customized: boolean = false) {
       this.pathToSvg = pathToSvg
-      this.outputPathToSvg = outputPathToSvg
-      this.colorCode = customizationConfig.colorCode!
-      this.rotateAngle = customizationConfig.rotateAngle!
-      this.flip = customizationConfig.flip!
+      this.colorCode = customizationConfig?.colorCode!
+      this.rotateAngle = customizationConfig?.rotateAngle!
+      this.flip = customizationConfig?.flip!
+      this.customized = customized
     }
 
     get getSvgCode () {
       return this.svgCode
     }
 
-    readSvgCode () {
+    private readSvgCode () {
       if (fs.existsSync(this.pathToSvg)) {
         this.svgCode = fs.readFileSync(this.pathToSvg, 'utf8')
       } else {
         this.svgCode = fs.readFileSync(this.pathToSvg.replace('svg-outlined', 'svg'), 'utf8')
       }
     }
 
-    modifySvgCode () {
+    private modifySvgCode () {
       const gTagOpen = `<g ${this.changeColor()} transform="${this.rotateIcon()} ${this.flipIcon()}">`
```

Comment:
So, here is an edge case (we can fix this later), in certain scenarios when rotate and flip both are applied, this customization wont work as we dont know what was applied first. (happens usually with asymmetric icons) but we should try to fix this later. Let's talk about this offline

Reply:
Hmm, ok, graphically, I think flip should be applied first in all cases, then we can make the rotation, I'm not really sure.